utils4e.py

```diff
@@ -277,33 +247,17 @@ def vector_add(a, b):
         try:
             return a + b
         except TypeError:
-            raise Exception("Inputs must be in the same size!")
-
-
-def scalar_vector_product(X, Y):
-    """Return vector as a product of a scalar and a vector recursively"""
-    return [scalar_vector_product(X, y) for y in Y] if hasattr(Y, '__iter__') else X * Y
+            raise Exception('Inputs must be in the same size!')
 
 
-def map_vector(f, X):
-    """apply function f to iterable X"""
-    return [map_vector(f, x) for x in X] if hasattr(X, '__iter__') else list(map(f, [X]))[0]
+def scalar_vector_product(x, y):
+    """Return vector as a product of a scalar and a vector recursively."""
+    return [scalar_vector_product(x, _y) for _y in y] if hasattr(y, '__iter__') else x * y
 
 
-def scalar_matrix_product(X, Y):
-    """Return matrix as a product of a scalar and a matrix"""
-    return [scalar_vector_product(X, y) for y in Y]
-
-
-def inverse_matrix(X):
```

Comment:
Same here again, what replaces these functions?

Reply:
Again, the same here. I replaced it with numpy.linalg.inv.